EDMarketConnector.py

```diff
@@ -11,91 +11,218 @@
 import webbrowser
 from builtins import object, str
 from os import chdir, environ
+from os import getpid as os_getpid
 from os.path import dirname, isdir, join
 from sys import platform
 from time import localtime, strftime, time
-from typing import TYPE_CHECKING, Any, Mapping, Optional, Tuple, cast
+from typing import TYPE_CHECKING, Any, Mapping, Optional, Tuple
 
 from constants import applongname, appname, protocolhandler_redirect
 
 # config will now cause an appname logger to be set up, so we need the
 # console redirect before this
 if __name__ == '__main__':
+    # Keep this as the very first code run to be as sure as possible of no
+    # output until after this redirect is done, if needed.
+    if getattr(sys, 'frozen', False):
+        # By default py2exe tries to write log to dirname(sys.executable) which fails when installed
+        import tempfile
+
+        # unbuffered not allowed for text in python3, so use `1 for line buffering
+        sys.stdout = sys.stderr = open(join(tempfile.gettempdir(), f'{appname}.log'), mode='wt', buffering=1)
+    # TODO: Test: Make *sure* this redirect is working, else py2exe is going to cause an exit popup
+
+# These need to be after the stdout/err redirect because they will cause
+# logging to be set up.
+# isort: off
+import killswitch
+from config import appversion, appversion_nobuild, config, copyright
```

Comment:
Do we want to switch some of these to be `constants` imports?

Reply:
At some point, yes.  But that would be for another PR.

In general this PR is just to get things in sync with what's in `main` for these changes.  I'd prefer your review be limited to finding any actual mistakes ;) .